test/timedemo_test.cpp

```diff
@@ -0,0 +1,71 @@
+#include <filesystem>
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "diablo.h"
+#include "engine/demomode.h"
+#include "options.h"
+#include "pfile.h"
+#include "utils/display.h"
+#include "utils/paths.h"
+
+using namespace devilution;
+
+namespace {
+
+bool Dummy_GetHeroInfo(_uiheroinfo *pInfo)
+{
+	return true;
+}
+
+void RunTimedemo(std::string timedemoFolderName)
+{
+	std::string unitTestFolderCompletePath = paths::BasePath() + "/test/fixtures/timedemo/" + timedemoFolderName;
+	paths::SetPrefPath(unitTestFolderCompletePath);
+	paths::SetConfigPath(unitTestFolderCompletePath);
+	LoadCoreArchives();
+	LoadGameArchives();
+
+	// The tests need spawn.mpq or diabdat.mpq
+	// Please provide them so that the tests can run successfully
+	ASSERT_TRUE(spawn_mpq || diabdat_mpq);
+
+	InitKeymapActions();
+	LoadOptions();
+
+	const int demoNumber = 0;
+
+	// Currently only spawn.mpq is present when building on github actions
+	gbIsSpawn = true;
+	gbIsHellfire = false;
+	gbMusicOn = false;
+	gbSoundOn = false;
+	HeadlessMode = true;
+	demo::InitPlayBack(demoNumber, true);
+
+	pfile_ui_set_hero_infos(Dummy_GetHeroInfo);
+	gbLoadGame = true;
+
+	demo::OverrideOptions();
+
+	AdjustToScreenGeometry(*sgOptions.Graphics.resolution);
+
+	StartGame(false, true);
+
+	ASSERT_EQ(pfile_compare_hero_demo(demoNumber), HeroCompareResult::Same);
+	ASSERT_FALSE(gbRunGame);
+	gbRunGame = false;
+	init_cleanup();
+}
+
+} // namespace
+
+TEST(Timedemo, RogueLevel1to2)
+{
+	RunTimedemo("RogueLevel1to2");

```

Comment:
Does this demo work? Spawn does not have the Rogue and I haven't ben able to load the save either.

Reply:
I tried it locally with spaw.mpq.
In normal gameplay it crashes, cause the animations are missing.
Vut with headless mode we don't need the animations and it worked.

Reply:
It always failed for me in headless

Reply:
Tested again, still works. 😵‍
I have no problem with removing them. They are a little bit hacky, cause rouge is normally not supported in spawn.
Anyway I'm still glad that the tests now run in github actions. 🙂

Reply:
Once merged you can open a new PR with more tests and if they work then we can merge :D